app/views/episodes/_form_media.html.erb

```diff
@@ -13,7 +13,11 @@
             <div class="form-floating" data-controller="dynamic-form">
               <%= form.select :medium, episode_medium_options, {}, data: {action: "dynamic-form#change"} %>
               <%= form.label :medium %>
-              <a hidden href="<%= request.fullpath %>" data-dynamic-form-target="link"></a>
+              <% if episode.new_record? %>
+                <a hidden href="<%= new_podcast_episode_path(episode.podcast_id) %>" data-dynamic-form-target="link"></a>
+              <% else %>
+                <a hidden href="<%= episode_path(episode) %>" data-dynamic-form-target="link"></a>
+              <% end %>
```

Comment:
This was a 500 bug creating a new episode.  If you submit with an error (leaving the segment count blank)... this magic had the wrong url.  It had `/podcast/123/episodes` instead of `/podcast/123/episodes/new`.